components/Markets/MarketDetailCard.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState } from 'react';
+import {
+  Text,
+  Group,
+  Loader,
+  Title,
+  Stack,
+  Card,
+  Divider,
+} from '@mantine/core';
+import { utf8 } from '@coral-xyz/anchor/dist/cjs/utils/bytes';
+import { quoteLotsToUi, baseLotsToUi } from '@openbook-dex/openbook-v2';
+import { useOpenbookMarket } from '@/contexts/OpenbookMarketContext';
+import { OrderBookCard } from '../OrderBook/OrderBookCard';
+import { OrderConfigurationCard } from '../OrderBook/OrderConfigurationCard';
+import DisableNumberInputScroll from '../Utilities/DisableNumberInputScroll';
+import { useExplorerConfiguration } from '@/hooks/useExplorerConfiguration';
+import { shortKey } from '@/lib/utils';
+import { MarketOrdersCard } from './MarketOrdersCard';
+
+export function MarketDetailCard() {
+  const openbookMarket = useOpenbookMarket();
+  const { generateExplorerLink } = useExplorerConfiguration();
+  const [price, setPrice] = useState<string>('');
+
+  const setPriceFromOrderBook = (value: string) => {
+    setPrice(value);
+  };
+
+  return openbookMarket.loading || !openbookMarket.market ? (
+    <Group justify="center">
+      <Loader />
+    </Group>
+  ) : (
+    <>
+      <Stack>
+        <Title>
+          {utf8.decode(new Uint8Array(openbookMarket.market.market.name)).split('\x00')[0]}
+        </Title>
+        <Text>Event Heap Size: {openbookMarket.eventHeapCount}</Text>
+        <Text>Event Heap Account:{' '}
+          <a
+            href={generateExplorerLink(openbookMarket.market.market.eventHeap.toString(), 'account')}
+          >
+            {shortKey(openbookMarket.market.market.eventHeap.toString())}
```

Comment:
Is this a computationally expensive function? If it is, we may want to memoize. If not, disregard.

Reply:
@R-K-H I don't think I have permission to resolve, so please resolve all my comments at your leisure.